File: Palindrome Binary.java
//Problem:  Palindrome Binary
//geeksforgeeks problem of the day -> 4th May 2026
//JAVA CODE
class Solution {
    static long isPalindrome(long N){
        long original=N;
        long rev=0;
        while(N>0){
            rev=(rev<<1)|(N&1);
            N=N>>1;
        }
        return (rev==original)?1:0;
    }
}

File: Palindrome Binary.cpp
//Problem:  Palindrome Binary
//geeksforgeeks problem of the day -> 4th May 2026
//C++ CODE

class Solution {
  public:
    int isPallindrome(long long int N) {
        // code here
        ll temp=N;
        ll rev=0;
        while(temp)
        {
            if(temp&1)
            {
                rev|=1;
            }
            temp>>=1;
            if(temp)
            {
                rev<<=1;
            }
        }
        return (N==rev);
    }
};